src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas1OutOfServiceBedEntity.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+
+@Repository
+interface Cas1OutOfServiceBedRepository : JpaRepository<Cas1OutOfServiceBedEntity, UUID>
+
+@Entity
+@Table(name = "cas1_out_of_service_beds")
+data class Cas1OutOfServiceBedEntity(
+  @Id
+  val id: UUID,
+  @ManyToOne
+  @JoinColumn(name = "premises_id")
+  val premises: ApprovedPremisesEntity,
+  @ManyToOne
+  @JoinColumn(name = "out_of_service_bed_reason_id")
+  val reason: Cas1OutOfServiceBedReasonEntity,
+  @ManyToOne
+  @JoinColumn(name = "bed_id")
+  val bed: BedEntity,
+  val createdAt: OffsetDateTime,
```

Comment:
how about using the ` @CreationTimestamp` annotation here to set this automatically? 

We're using this on several entities already. I tend to think that having automatic `created_at` and `updated_at` timestamps should be standard hygiene on all tables and would be v happy to see that here.

https://www.baeldung.com/hibernate-creationtimestamp-updatetimestamp

Reply:
I think we'd still have to pass in an `OffsetDateTime` when constructing an instance anyway, so I'm not sure what we gain to make up for making the origin of the timestamp less explicit.